### PR TITLE
Do not trigger preview hooks if enable_previews is false

### LINKED ISSUE #15784: enable_previews => false does not cancel preview hooks
With the config option `enable_previews=>false`, the hooks for the previews are still called.
An extra check should we putted here https://github.com/owncloud/core/blob/1ee2ee84327a23ebf089f5b719f871168d6d007d/lib/base.php#L763, I'll create a pull request.

### PR DESCRIPTION
**Pull request by labkode**
Closes #15784

### CODE REVIEW

**Comment by nickvergessen** [comment]
@owncloud-bot it's okay to test this

File: lib/base.php [hunk 1]
```diff
@@ -766,13 +766,15 @@ public static function registerFilesystemHooks() {
...
 	 */
 	public static function registerPreviewHooks() {
-		OC_Hook::connect('OC_Filesystem', 'post_write', 'OC\Preview', 'post_write');
-		OC_Hook::connect('OC_Filesystem', 'delete', 'OC\Preview', 'prepare_delete_files');
-		OC_Hook::connect('\OCP\Versions', 'preDelete', 'OC\Preview', 'prepare_delete');
-		OC_Hook::connect('\OCP\Trashbin', 'preDelete', 'OC\Preview', 'prepare_delete');
-		OC_Hook::connect('OC_Filesystem', 'post_delete', 'OC\Preview', 'post_delete_files');
-		OC_Hook::connect('\OCP\Versions', 'delete', 'OC\Preview', 'post_delete');
-		OC_Hook::connect('\OCP\Trashbin', 'delete', 'OC\Preview', 'post_delete');
+		if (\OC::$server->getSystemConfig()->getValue('enable_previews', false)) {
```
Comment (nickvergessen) [inline lib/base.php]: I think we should wrap the call to this method in line 657 instead of the complete content:
```
self::registerPreviewHooks();
```

**Comment by labkode** [comment]
I moved the check to init()

**Comment by DeepDiver1975** [comment]
@owncloud-bot this is okay to test

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
:+1: THX a lot @labkode

**Comment by nickvergessen** [comment]
@carlaschroder not sure if it is documented, so I just want to check back. We decided to not slow down oc unnecessarily. So we will not clear thumbnails, when an admin changes the setting.
This needs to be done manually (in the data dir, remove all `thumbnails` folders inside the user directories). This should be acceptable, because changing this setting more then once should not be very common.

**Comment by nickvergessen** [comment]
:+1:

**Pull request merged**